destination/format/csv.go

```diff
@@ -0,0 +1,188 @@
+// Copyright © 2024 Meroxa, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package format
+
+import (
+	"bytes"
+	"encoding/csv"
+	"encoding/json"
+	"fmt"
+	"time"
+
+	sdk "github.com/conduitio/conduit-connector-sdk"
+	"github.com/go-errors/errors"
+	"golang.org/x/exp/maps"
+)
+
+func makeCSVRecords(records []sdk.Record, prefix string, orderingColumns []string) (
+	*bytes.Buffer, *bytes.Buffer, map[string]string, []string, []string, error,
```

Comment:
This is a place where name returns will make the code more readable.

https://go.dev/tour/basics/7

Reply:
Adding to alternatively as well... you can pass the response as a parameter, allowing the called to cache (using a `sync.Pool`) the response and reuse the objects. _(only worth it if you can have a situation of not knowing the size of the data)_

Reply:
@nassor the better alternative is to not have more than 2 returns. We are abusing the language here.